src/tree/mod.rs

```diff
@@ -742,13 +712,18 @@ impl Tree {
         // https://fjall-rs.github.io/post/bloom-filter-hash-sharing/
         let key_hash = crate::table::filter::standard_bloom::Builder::get_hash(key);
 
+        let mut best: Option<InternalValue> = None;
+
         for level in version.iter_levels() {
             for run in level.iter() {
                 // NOTE: Based on benchmarking, binary search is only worth it with ~4 tables
                 if run.len() >= 4 {
                     if let Some(table) = run.get_for_key(key) {
                         if let Some(item) = table.get(key, seqno, key_hash)? {
-                            return Ok(ignore_tombstone_value(item));
```

Comment:
This will make a point read cascade through all levels, even though through compaction we guarantee that the first item we find (in the "most upper" run), will have the highest version, no?

Reply:
that's correct, that was my mistake and should be fixed now.